ev3dev2/motor.py

```diff
@@ -1921,6 +1954,96 @@ def on(self, left_speed, right_speed):
         self.left_motor.run_forever()
         self.right_motor.run_forever()
 
+    def follow_line(self,
+            kp, ki, kd,
+            speed, keep_following_function,
+            target_light_intensity=None,
+            follow_left_edge=True,
+            WHITE=60,
+            OFF_LINE_COUNT_MAX=20,
+            SLEEP_TIME=0.01
+        ):
+        """
+        PID line follower
+
+        ``kp``, ``ki``, and ``kd`` are the PID constants.
+
+        ``speed`` is the desired speed of the midpoint of the robot
+
+        ``keep_following_function`` is called to determine if we should keep
```

Comment:
What are the expected scenarios here? I'd imagine the big ones are "follow for a distance" and "follow for a time"; could we provide easy ways to do that? Either provide functions which return the appropriate predicate, or (my preferred solution) also have keyword parameters for `follow_time` and `follow_distance`. And perhaps specifying none of them defaults to going forever.

We should also have an example of how to use this to specify common conditions.

Reply:
We could provide some basic `follow_` functions here in motor.py:
- `follow_forever`
- `follow_for_ms`
- `follow_for_distance`

Where the user passes in the name of the function for the one they wish to use.  We need to make this flexible though...am thinking of the FLL scenario where they need to do something more complicated like `follow_until_intersection_with_another_line`.

Reply:
Actually we cannot do `follow_for_distance` here because this code is all in `MoveTank` where we do not know the wheel dimensions.  The user could always do a custom `follow_for_` function where they implement that.